src/cljs/org/broadinstitute/firecloud_ui/page/library/library_page.cljs

```diff
@@ -73,12 +74,12 @@
                         (map #((keyword %) data)
                              (concat [:library:indication :library:dataUseRestriction :library:numSubjects]
                                      extra-columns))))}]))
-   :component-will-receive-props
-   (fn [{:keys [next-props refs]}]
-     (let [current-search-text (:filter-text (react/call :get-query-params (@refs "table")))
-           new-search-text (:search-text next-props)]
-       (when-not (= current-search-text new-search-text)
-         (react/call :update-query-params (@refs "table") {:filter-text new-search-text}))))
+   :execute-search
+   (fn [{:keys [refs]}]
+     (when (and (not (nil? (@refs "table")))
+                (= (:current-page (react/call :get-query-params (@refs "table"))) 1))
+       (react/call :execute-search (@refs "table"))
+       (react/call :update-query-params (@refs "table") {:current-page 1})))
```

Comment:
Similar issue is in play with the search bar. Navigate to page 2, enter something that would normally filter the results, and nothing happens.